### DISCUSSION TITLE
Is there a way to validate only one field at a time, instead of the entire form?

### DISCUSSION THREAD

**Discussion by Ryan-Nunes** [react-hook-form/react-hook-form#3699] [Q&A] [answered]
Currently with RHF, I have a yup validation schema which, for one of the input fields in my form, makes a call to my backend to validate the value. For performance reasons, I only want to call this validation endpoint when the user changes that one specific field, however it seems that when the user changes any of the fields in my form, the validation for all fields is run at the same time.

Is there a way to make RHF only validate the field which the user is interacting with?

**Comment by bluebill1049** [comment, marked as answer]
That's the natural behavior when working with schema lib like yup, we are potentially looking into providing field name which is validating and let the scheme validation (validate single input) if they have the capability.

**Comment by mic4ael** [comment]
That is something I would really appreciate. Is there a workaround for this?

**Reply by Ryan-Nunes** [reply]
Currently, what I do is pass the Yup validation schema and the field name to a method which then uses the yup schema functions to validate my specific field.

**Reply by azabraao** [reply]
@Ryan-Nunes can you demonstrate that?

**Reply by Ryan-Nunes** [reply]
I unfortunately cannot create a full demo for you because the code I have to base myself off of is private, but can refer you to this documentation for yup validation that I used to make the code: https://github.com/jquense/yup#mixedvalidateatpath-string-value-any-options-object-promiseany-validationerror

**Reply by azabraao** [reply]
The example works fine. Thanks!

**Reply by AuthorProxy** [reply]
@Ryan-Nunes your link is not working, can you paste a part of the code?

**Reply by azabraao** [reply]
@AuthorProxy it still there in the history:

https://github.com/jquense/yup/blob/0fca0a49c7305a22dc264d9f42c11040dd63a491/README.md#mixedvalidateatpath-string-value-any-options-object-promiseany-validationerror

**Reply by AuthorProxy** [reply]
> @AuthorProxy it still there in the history:
> 
> https://github.com/jquense/yup/blob/0fca0a49c7305a22dc264d9f42c11040dd63a491/README.md#mixedvalidateatpath-string-value-any-options-object-promiseany-validationerror

thks, I made it using manuall calling `trigger` from RHF for now
